src/SourceBuild/content/eng/tools/init-build.proj

```diff
@@ -67,17 +67,34 @@
              Properties="MSBuildRestoreSessionId=$([System.Guid]::NewGuid())" />
 
     <MSBuild Projects="tasks\SourceBuild.MSBuildSdkResolver\SourceBuild.MSBuildSdkResolver.csproj"
-             Targets="Build;InstallResolver" />
+             Targets="Build">
+      <Output TaskParameter="TargetOutputs" ItemName="SourceBuildSdkResolver" />
+    </MSBuild>
 
     <MakeDir Directories="$(BaseIntermediateOutputPath)" />
     <Touch Files="$(BaseIntermediateOutputPath)BuildMSBuildSdkResolver.complete" AlwaysCreate="true">
       <Output TaskParameter="TouchedFiles" ItemName="FileWrites" />
     </Touch>
   </Target>
 
+  <PropertyGroup>
+    <SourceBuildMSBuildSdkResolverManifestPath>$([MSBuild]::NormalizePath('$(VSMSBuildSdkResolversDir)', 'SourceBuild.MSBuildSdkResolver', 'SourceBuild.MSBuildSdkResolver.xml'))</SourceBuildMSBuildSdkResolverManifestPath>
+  </PropertyGroup>
+
+  <Target Name="LayoutMSBuildSdkResolver"
+          DependsOnTargets="BuildMSBuildSdkResolver"
+          Inputs="$(MSBuildProjectFullPath)"
+          Outputs="$(SourceBuildMSBuildSdkResolverManifestPath)">
+
+    <WriteLinesToFile
+      File="$(SourceBuildMSBuildSdkResolverManifestPath)"
+      Lines="&lt;SdkResolver&gt;&lt;Path&gt;%(SourceBuildSdkResolver.FullPath)&lt;/Path&gt;&lt;/SdkResolver&gt;"
+      Overwrite="true" />
+  </Target>
```

Comment:
What was your reason to put this logic into a separate target and outside of the MSBuildSdkResolver.csproj? IMO it would be fine it the project would product this artifacts as part of the build.

Reply:
Initially when I thought the environment variables worked differently,there was a significant amount of logic here to copy the existing resolvers. Now that it's just this one, we could have it lay itself out. In general though, I thought we were avoiding using non-standard entry-point target names due to aspirations to use static graph (or is this only in the runtime repo?)

Reply:
We don't need it to be an entry point target in `SourceBuild.MSBuildSdkResolver.csproj`. I would just sequence it into the build, i.e. to run before/after the `PrepareForRun` target. Static graph concerns only apply to entry point targets.